ci/test/00_setup_env_native_nowallet.sh

```diff
@@ -8,7 +8,7 @@ export LC_ALL=C.UTF-8
 
 export CONTAINER_NAME=ci_native_nowallet
 export DOCKER_NAME_TAG=ubuntu:18.04  # Use bionic to have one config run the tests in python3.6, see doc/dependencies.md
-export PACKAGES="python3-zmq clang-5.0 llvm-5.0"  # Use clang-5 to test C++17 compatibility, see doc/dependencies.md
-export DEP_OPTS="NO_WALLET=1"
+export PACKAGES="python3-zmq clang-7 llvm-7 libc++abi-7-dev libc++-7-dev"  # Use clang-7 to test C++17 compatibility, see doc/dependencies.md
+export DEP_OPTS="NO_WALLET=1 CC=clang-7 CXX='clang++-7 -stdlib=libc++'"
```

Comment:
182de7ba10811ec39e24ec5bec7cd2119f776f2f

Passing `CC`and `CXX` to `DEP_OPTS` does not work as expected. From the [log](https://api.cirrus-ci.com/v1/task/5298955543117824/logs/ci.log):
```
...
Configuring qt...
...

Configure summary:

Building on: linux-g++ (x86_64, CPU features: mmx sse sse2)
Building for: linux-g++-64 (x86_64, CPU features: mmx sse sse2)
Target compiler: gcc 7.5.0
...
```

Reported in #22184. 

---

Another issue:
```
...
checking for QMinimalIntegrationPlugin (-lqminimal)... no
configure: WARNING: QMinimalIntegrationPlugin not found.; bitcoin-qt frontend will not be built
...

Options used to compile and link:
  external signer = yes
  multiprocess    = no
  with libs       = yes
  with wallet     = no
  with gui / qt   = no
...
```

Reported in #22344, fixed in #22814, some relevant discussion in https://github.com/bitcoin/bitcoin/pull/22815#discussion_r697489586.

Reply:
> Passing CC and CXX to DEP_OPTS does not work as expected. 

It works as expected with everything except for Qt, and that's not an issue introduced by this PR.

> Reported in #22344, fixed in #22814, some relevant discussion in

Another Qt specific issue, that isn't introduced by this PR.

Frankly I'm getting sick of fighting against Qts build system, and don't consider either of this issues blockers for this change, especially if there's a potential fix available. I'll comment in #22814 shortly.

Reply:
> ... isn't introduced by this PR.

This PR builds Qt dependencies with the `-stdlib=libc++`. It was not the case before.

Reply:
> This PR builds dependencies with the -stdlib=libc++. It was not the case before.

The underlying issue, which is that the Qt build doesn't work properly in regards to building with ` -stdlib=libc++`, was not introduced in this PR. I don't think it's an issue that we have one less CI building Qt in depends.